test/vcf/assembly_checker_integration_test.cpp

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "assembly_checker_test_aux.hpp"
+
+namespace ebi
+{  
+
+  TEST_CASE("Valid fasta and vcf combination", "[assembly_checker]")
+  {
+      SECTION("Empty VCF File")
+      {
+          std::string folder = "test/input_files/assembly_checker/passed/passed_empty_vcf/";
+          std::string file_prefix = get_file_prefix(folder);
+          CHECK(ebi::vcf::assembly_checker::check_vcf_ref(folder + file_prefix + ebi::vcf::VCF_EXT, folder + file_prefix + ebi::vcf::FASTA_EXT, folder + file_prefix + ebi::vcf::INDEX_EXT));
+      }
+      SECTION("Single entry")
+      {
+          std::string folder = "test/input_files/assembly_checker/passed/passed_single_variant/";
+          std::string file_prefix = get_file_prefix(folder);
+          CHECK(ebi::vcf::assembly_checker::check_vcf_ref(folder + file_prefix + ebi::vcf::VCF_EXT, folder + file_prefix + ebi::vcf::FASTA_EXT, folder + file_prefix + ebi::vcf::INDEX_EXT));
+      }
+      SECTION("Full sample VCF, all match")
+      {
+          std::string folder = "test/input_files/assembly_checker/passed/passed_fullsample/";
+          std::string file_prefix = get_file_prefix(folder);
+          CHECK(ebi::vcf::assembly_checker::check_vcf_ref(folder + file_prefix + ebi::vcf::VCF_EXT, folder + file_prefix + ebi::vcf::FASTA_EXT, folder + file_prefix + ebi::vcf::INDEX_EXT));
+      }
+  }
+
+  TEST_CASE("Invalid VCF and Fasta combination", "[assembly_checker]")
+  {
+      SECTION("Single entry, single mismatch")
+      {
+          std::string folder = "test/input_files/assembly_checker/failed/failed_singlemismatch_singleentry/";
+          std::string file_prefix = get_file_prefix(folder);
+          CHECK_FALSE(ebi::vcf::assembly_checker::check_vcf_ref(folder + file_prefix + ebi::vcf::VCF_EXT, folder + file_prefix + ebi::vcf::FASTA_EXT, folder + file_prefix + ebi::vcf::INDEX_EXT));
```

Comment:
I like this test suite a lot more now :) You can check [this](https://github.com/EBIvariation/vcf-validator/blob/master/test/vcf/parser_v43_test.cpp#L52-L65) and [this](https://github.com/EBIvariation/vcf-validator/blob/master/test/vcf/parser_test_aux.hpp#L34-L41) on how to test the entry point method using streams instead of file paths as arguments.

Note: In fact one of the methods linked receives a file path but that could be stdin.